pypdf/_page.py

```diff
@@ -444,19 +444,78 @@ def _get_ids_image(
-                continue
-            if x_object[o][IA.SUBTYPE] == "/Image":
-                lst.append(o if len(ancest) == 0 else ancest + [o])
-            else:  # is a form with possible images inside
-                lst.extend(self._get_ids_image(x_object[o], ancest + [o], call_stack))
+
+        if PG.ANNOTS in obj:
+            for annot in cast(DictionaryObject, obj[PG.ANNOTS]):
+                if (
+                    "/AP" in cast(DictionaryObject, annot.keys())
+                    and "/N" in cast(DictionaryObject, annot["/AP"].keys())
+                    and PG.RESOURCES in annot["/AP"]["/N"].get_object()
+                    and RES.XOBJECT
+                    in cast(DictionaryObject, annot["/AP"]["/N"][PG.RESOURCES])
+                    and "/FRM"
+                    in cast(
+                        DictionaryObject, annot["/AP"]["/N"][PG.RESOURCES][RES.XOBJECT]
+                    )
+                ):
+                    frame = annot["/AP"]["/N"][PG.RESOURCES][RES.XOBJECT]["/FRM"]
+
+                    if PG.RESOURCES in frame.get_object() and RES.XOBJECT in cast(
+                        DictionaryObject, frame[PG.RESOURCES]
+                    ):
+                        x_object = frame[PG.RESOURCES][RES.XOBJECT]
+                        for o in x_object:
+                            if (
+                                isinstance(x_object[o], StreamObject)
+                                and x_object[o][IA.SUBTYPE] == "/Image"
+                            ):
+                                lst.extend(
+                                    [
+                                        (
+                                            f"{PG.ANNOTS}/{annot['/T']}{o}"
+                                            if len(ancest) == 0
+                                            else ancest + [o]
+                                        )
+                                    ]
+                                )
+
+        if PG.RESOURCES in obj:
+            if RES.PATTERN in cast(DictionaryObject, obj[PG.RESOURCES]):
+                for pattern_name, pattern in cast(
+                    DictionaryObject,
+                    cast(DictionaryObject, obj[PG.RESOURCES])[RES.PATTERN],
+                ).items():
+                    if PG.RESOURCES in pattern.get_object() and RES.XOBJECT in cast(
+                        DictionaryObject, pattern[PG.RESOURCES]
+                    ):
+                        x_object = pattern[PG.RESOURCES][RES.XOBJECT].get_object()
+                        for o in x_object:
+                            if (
+                                isinstance(x_object[o], StreamObject)
+                                and x_object[o][IA.SUBTYPE] == "/Image"
+                            ):
+                                lst.extend(
+                                    [
+                                        (
+                                            f"{RES.PATTERN}{pattern_name}{o}"
```

Comment:
for the name as proposed above I woud come to 
reader.pages[0].images['Pattern’,'/Img']